[CHIP-TOOL] Error occurs while compiling CHIP-TOOL

### Reproduction steps

1. gn gen out/debug --args='chip_mdns="platform" chip_inet_config_enable_ipv4=false'
2. ninja -C out/debug

### Bug prevalence

After the PR : https://github.com/project-chip/connectedhomeip/pull/27839

### GitHub hash of the SDK that was being used

9bf0fe6c63979701b4344d2fce1a54ca8bdbf0a3

### Platform

raspi

### Platform Version(s)

RPI-4, 8GB-RAM variant

### Error Log :

```
ninja: Entering directory `out/debug'
[619/2140] c++ obj/src/lib/format/tests/libFormatTests.TestDecoding.cpp.o
FAILED: obj/src/lib/format/tests/libFormatTests.TestDecoding.cpp.o
aarch64-linux-gnu-g++ -MMD -MF obj/src/lib/format/tests/libFormatTests.TestDecoding.cpp.o.d -Wconversion -O0 -g2 -fno-common -ffunction-sections -fdata-sections -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -Wall -Werror -Wextra -Wshadow -Wunreachable-code -Wvla -Wformat -Wformat-nonliteral -Wformat-security -Wno-deprecated-declarations -Wno-missing-field-initializers -Wno-unknown-warning-option -Wno-unused-parameter -Wno-cast-function-type -Wno-psabi -Wno-maybe-uninitialized -fdiagnostics-color -fno-strict-aliasing -fmacro-prefix-map=../../= -std=gnu++14 -fno-rtti -Wnon-virtual-dtor -DCHIP_HAVE_CONFIG_H=1 -Igen/src/controller/data_model -I../../src/include -I../../src -Igen/include -I../../zzz_generated/app-common -I../../config/standalone -I../../third_party/nlassert/repo/include -I../../third_party/nlio/repo/include -I../../third_party/nlfaultinjection/repo/include -Igen/src/lib/format -I../../third_party/nlunit-test/repo/src -c ../../src/lib/format/tests/TestDecoding.cpp -o obj/src/lib/format/tests/libFormatTests.TestDecoding.cpp.o
../../src/lib/format/tests/TestDecoding.cpp: In function ‘void __static_initialization_and_destruction_0(int, int)’:
../../src/lib/format/tests/TestDecoding.cpp:37:23: error: statement has no effect [-Werror=unused-value]
   37 | const Entry<ItemInfo> _empty_item[0]                 = {};
      |                       ^~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[624/2140] c++ obj/src/lib/shell/shell_core.Engine.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/src/lib/format/tests/TestDecoding.cpp
+++ b/src/lib/format/tests/TestDecoding.cpp
@@ -34,7 +34,12 @@ using namespace chip::TLV;
 using namespace chip::TLVMeta;
 using namespace chip::TestData;
 
-const Entry<ItemInfo> _empty_item[0]                 = {};
+// size 1 to avoid compilers complaining about empty arrays
+// (not allowed by ISO 9899:2011 6.7.6.2:
+//    If the expression is a constant expression, it shall have a value greater
+//    than zero.
+// ). We still claim its size is 0 in empty_meta though.
+const Entry<ItemInfo> _empty_item[1]                 = {};
 const std::array<const Node<ItemInfo>, 1> empty_meta = { { { 0, _empty_item } } };
 
 const Entry<ItemInfo> _FakeProtocolData[] = {

